syntax = "proto3";
package user;
option go_package = "/proto";

import "src/api/common.proto";

message JwtTokenPayload {
  string token = 1;
  int32 ttl = 2;
}

message SignUpRequest {
  string email = 1;
  string password = 2;
}

message SignUpResponse {
  JwtTokenPayload payload = 1;
}

message UsersInternalRequest {
  common.Pagination pagination = 1;
}

message UsersInternalResponse {
  repeated common.UserProfile availableUsers = 1;
}

service UserService {
  rpc SignUp (SignUpRequest) returns (SignUpResponse);
  rpc SignIn (SignUpRequest) returns (SignUpResponse);
  rpc Profile (common.Empty) returns (common.UserProfile);
  rpc UsersInternal (UsersInternalRequest) returns (UsersInternalResponse);
}
